### DISCUSSION TITLE
BLE multiple sensors

### DISCUSSION THREAD

**Discussion by MrYoshii** [syssi/esphome-jk-bms#133] [Q&A] [answered]
Hi I have a JK-B2A8S20P and use an ESP32S for BLE.
I only have 8S so I have deleted the sensors for the remaining cells. 
But I would like to add more sensors. Is this possible?
Would like to read the ballance stom and the cell difference.

Thanks for your help.

**Comment by syssi** [comment, marked as answer]
Could you provide a ESPHome log to illustrate the issue? You could simply add the balance current sensor:

```
    balancing_current:
      name: "${name} balancing current"
```

**Reply by MrYoshii** [reply]
Thank you for the quick reply. I found out that the sensor was already in there and I just overlooked it. Is there a list of all available sensors?
Greetings

**Reply by syssi** [reply]
The example contains all available sensors: https://github.com/syssi/esphome-jk-bms/blob/main/esp32-ble-example.yaml

**Reply by MrYoshii** [reply]
Ah ok I thought there are more because in the other example configs are other sensors. 
Then that was it thanks.

**Reply by syssi** [reply]
This component does implement different protocols:

1. BLE JK04
2. BLE JK02
3. UART-TTL

All protocols have different payloads / exposing different informations. F.e. the balancing current isn't available via UART-TTL. Using the BLE protocol is the best choice because it's the only way to control the BMS. UART-TTL is read-only.
